### ISSUE TITLE
training is so slow

### ISSUE THREAD

**Issue by olfa-koubaa** [open-mmlab/mmdetection#5237]
I tried training the mask rcnn with both detectron2 and mmdetection and the training speed gap is huge 

mmdetection offers a lot of preimplemented models that I want to run my tests with but I'm struggling with the fact that is so slow 

I found some people who encountered the same problem but it's not clear as to what's causing it or what is the solution

**Comment by RangiLyu** [comment]
Hi, can you provide more details e.g. training config, log files, and the training time difference to help us find out the reason?

**Comment by tehkillerbee** [comment]
Are you sure you are loading a checkpoint to speed up training? How many images are you using when training?

**Comment by olfa-koubaa** [comment]
> Are you sure you are loading a checkpoint to speed up training? How many images are you using when training?

yes I am 
I'm using 987 images for training which is not a lot so it shouldn't slow the training this much

**Comment by jshilong** [comment]
Different from Detectron2, In our implementation,  `dataloader` would relaunch after every epoch,  It would cost much time when your train a small dataset, We recommend you use `RepeatDataset` to build a larger dataset instead of setting a larger epoch number.

**Comment by jshilong** [comment]
We should add this feature to a future version. @RangiLyu @ZwwWayne @hhaAndroid

**Comment by olfa-koubaa** [comment]
> Different from Detectron2, In our implementation, `dataloader` would relaunch after every epoch, It would cost much time when your train a small dataset, We recommend you use `RepeatDataset` to build a larger dataset instead of setting a larger epoch number.

Thank you for your response

I'm using repeat dataset now in the traininig,validation and test datasets but I'm getting this error:


'RepeatDataset' object has no attribute 'evaluate'

should I use repeat dataset only in the training dataset ?

**Comment by RangiLyu** [comment]
Yes, you only need to use RepeatDataset in your training dataset.
